rackcity/views/it_model_views.py

```diff
@@ -169,22 +170,25 @@ def model_page(request):
             status=HTTPStatus.BAD_REQUEST,
         )
 
-    if 'sort_by' in request.data:
-        sort_by = request.data['sort_by']
-        sort_args = []
-        for sort in sort_by:
-            if ('field' not in sort) or ('ascending' not in sort):
-                failure_message += "Must specify 'field' and 'ascending' fields. "
-                return JsonResponse(
-                    {"failure_message": failure_message},
-                    status=HTTPStatus.BAD_REQUEST,
-                )
-            field_name = sort['field']
-            order = "-" if not sort['ascending'] else ""
-            sort_args.append(order + field_name)
-        models = ITModel.objects.order_by(*sort_args)
-    else:
-        models = ITModel.objects.all()
+    models_query = ITModel.objects
+
+    try:
+        filter_args = get_filter_arguments(request.data)
+    except Exception as error:
+        return JsonResponse(
+            {"failure_message": "Filter error: " + str(error)},
+            status=HTTPStatus.BAD_REQUEST
+        )
+    models_query = models_query.filter(**filter_args)
+
+    try:
+        sort_args = get_sort_arguments(request.data)
+    except Exception as error:
```

Comment:
I ran into a problem doing something like this where I got an unexpected exception (not the one I was intentionally raising in the method) because something went wrong in the method, but caught it thinking it was mine. Then I ended up displaying to the user something like "Sort error: couldn't find key 'id' in model_data"

I think better practice here is to make our own exception types and then say `except SortArgsException as error:`. Check out #81 for an example

Approving anyway bc that doesn't need to happen now

Reply:
ok swagging will keep this in mind when cleaning up code